File: Merge Sorted Array.cpp
class Solution {
public:
    void merge(int A[], int m, int B[], int n) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        int A_counter = m-1;
        int B_counter = n-1;
        
        for(int i = m+n-1; i > -1; i--)
        {
            int numA = (A_counter>-1)?A[A_counter]:INT_MIN;
            int numB = (B_counter>-1)?B[B_counter]:INT_MIN;
            if(numA > numB)
            {
                A[i] = numA;
                A_counter--;
            }
            else
            {
                A[i] = numB;
                B_counter--;
            }
        }
    }
};

File: Merge Sorted Array.java
public class Solution {
    public void merge(int A[], int m, int B[], int n) {
        // Start typing your Java solution below
        // DO NOT write main() function
        int countA = m-1;
        int countB = n-1;
        for(int i = m + n -1; i> -1; i--)
        {
            int a = (countA > -1)? A[countA]:Integer.MIN_VALUE;
            int b = (countB > -1)? B[countB]:Integer.MIN_VALUE;
            if(a > b)
            {
                A[i] = a;
                countA--;
            }
            else
            {
                A[i] = b;
                countB--;
            }
        }
    }
}